staging/src/k8s.io/component-base/config/testing/roundtrip.go

```diff
@@ -90,3 +92,89 @@ func GetRoundtripTestCases(scheme *runtime.Scheme, disallowMarshalGroupVersions
 	}
 	return cases
 }
+
+func runRoundTripTests(t *testing.T, tests []TestCase, codecs serializer.CodecFactory) {
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			object := decodeYAML(t, tc.in, tc.codec)
+
+			roundTrip(t, tc.codec, object)
+
+			data, err := runtime.Encode(tc.codec, object)
+			if err != nil {
+				t.Fatalf("failed to encode object: %v", err)
+				return
+			}
+
+			matchOutputFile(t, tc.out, data)
+		})
+	}
+}
+
+func roundTrip(t *testing.T, codec runtime.Codec, object runtime.Object) {
+	original := object
+
+	// deep copy the original object
+	object = original.DeepCopyObject()
+	if !apiequality.Semantic.DeepEqual(original, object) {
+		t.Errorf("deepcopy altered the original object, diff (- want, + got):\n%v", cmp.Diff(original, object))
+		return
+	}
+
+	// encode (serialize) the deep copy using the provided codec
+	data, err := runtime.Encode(codec, object)
+	if err != nil {
+		t.Errorf("failed to encode object: %v", err)
+		return
+	}
+
+	// ensure that the deep copy is equal to the original; neither the deep
+	// copy or conversion should alter the object
+	if !apiequality.Semantic.DeepEqual(original, object) {
+		t.Errorf("encode altered the object, diff (- want, + got): \n%v", cmp.Diff(original, object))
+		return
+	}
+
+	// encode (serialize) a second time to verify that it was not varying
+	secondData, err := runtime.Encode(codec, object)
+	if err != nil {
+		t.Errorf("failed to encode object: %v", err)
+		return
+	}
+
+	// serialization to the wire must be stable to ensure that we don't write twice to the DB
+	// when the object hasn't changed.
+	if !bytes.Equal(data, secondData) {
+		t.Errorf("serialization is not stable")
+		return
+	}
+
+	// decode (deserialize) the encoded data back into an object
+	obj2, err := runtime.Decode(codec, data)
+	if err != nil {
+		t.Errorf("failed to decode: %v", err)
+		return
+	}
+
+	// ensure that the object produced from decoding the encoded data is equal
+	// to the original object
+	if !apiequality.Semantic.DeepEqual(original, obj2) {
+		t.Errorf("decoding altered the object, diff (- want, + got):\n%v", cmp.Diff(object, obj2))
```

Comment:
Would it be clearer to say something like "object was not the same after round trip"?